forch/faucetizer.py

```diff
@@ -144,24 +148,38 @@ def _augment_include_file_name(self, file_name):
         base_file_name, ext = os.path.splitext(file_name)
         return base_file_name + INCLUDE_FILE_SUFFIX + ext
 
+    def _validate_and_initialize_config(self):
+        if self._config.fot_config.testing_segment:
+            starting_vlan = self._config.fot_config.testing_vlan_start
+            ending_vlan = self._config.fot_config.testing_vlan_end
+            if not starting_vlan or not ending_vlan:
+                raise Exception(
+                    f'Starting or ending testing VLAN missing: {starting_vlan}, {ending_vlan}')
+
+            testing_vlans = set(range(starting_vlan, ending_vlan+1))
+            operational_vlans = set(self._segments_to_vlans.values())
+
+            if testing_vlans & operational_vlans:
+                LOGGER.error(
+                    'Testing VLANs has intersection with operational VLANs: %s',
+                    testing_vlans & operational_vlans)
+
+            self._all_testing_vlans = testing_vlans - operational_vlans
+
     def _get_port_type(self, port_cfg):
-        testing_port_identifier = (self._config.orch_testing_config.testing_port_identifier or
+        testing_port_identifier = (self._config.fot_config.testing_port_identifier or
                                    TESTING_PORT_IDENTIFIER_DEFAULT)
-        if testing_port_identifier in port_cfg.get('description'):
-            return 'testing'
+        if testing_port_identifier in port_cfg.get('description', ""):
+            return PortType.testing
         non_access_port_properties = ['stack', 'lacp', 'output_only', 'tagged_vlans']
         port_properties = [
             property for property in non_access_port_properties if property in port_cfg]
-        return 'access' if len(port_properties) == 0 else 'other'
+        return PortType.access if len(port_properties) == 0 else PortType.unknown
```

Comment:
'unknown' is not the same as 'other' -- in this case, I think you do want 'other' (since it's not testing or access), while 'unknown' would be 0 or None (unassigned). So, the proto should have both, and this use should be 'other'.

Reply:
Done